kv/kv.go

```diff
@@ -0,0 +1,148 @@
+package kv
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/jordan-rash/wasmcloud-go/models"
+	"github.com/nats-io/nats.go"
+	core "github.com/wasmcloud/interfaces/core/tinygo"
+)
+
+const (
+	LINKDEF_PREFIX = "LINKDEF_"
+	CLAIMS_PREFIX  = "CLAIMS_"
+	BUCKET_PREFIX  = "LATTICEDATA_"
+)
+
+func GetKVStore(nc *nats.Conn, latticePrefix, jsDomain string) (nats.KeyValue, error) {
+	var js nats.JetStreamContext
+	var err error
+
+	if jsDomain != "" {
+		js, err = nc.JetStream(nats.PublishAsyncMaxPending(256), nats.Domain(jsDomain))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		js, err = nc.JetStream(nats.PublishAsyncMaxPending(256))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bucket := fmt.Sprintf("%s%s", BUCKET_PREFIX, latticePrefix)
+
+	return js.KeyValue(bucket)
+}
+
+func GetClaims(store nats.KeyValue) (*models.GetClaimsResponse, error) {
+	claims := models.CtlKVList{}
+	entries, err := store.Keys()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range entries {
+		if strings.HasPrefix(c, CLAIMS_PREFIX) {
+			entry, err := store.Get(c)
+			if err != nil {
+				return nil, err
+			}
+
+			d := models.KeyValueMap{}
+			err = json.Unmarshal(entry.Value(), &d)
+			if err != nil {
+				return nil, err
+			}
+
+			claims = append(claims, d)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return &models.GetClaimsResponse{Claims: claims}, nil
+}
+
+func GetLinks(store nats.KeyValue) (*models.LinkDefinitionList, error) {
+	links := core.ActorLinks{}
+	entries, err := store.Keys()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range entries {
+		if strings.HasPrefix(c, LINKDEF_PREFIX) {
+			entry, err := store.Get(c)
+			if err != nil {
+				return nil, err
+			}
+
+			d := core.LinkDefinition{}
+			err = json.Unmarshal(entry.Value(), &d)
+			if err != nil {
+				return nil, err
+			}
+
+			links = append(links, d)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return &models.LinkDefinitionList{Links: links}, nil
+}
+
+func PutLink(store nats.KeyValue, ld core.LinkDefinition) error {
+	id, err := LDHash(&ld)
+	if err != nil {
+		return err
+	}
+	key := fmt.Sprintf("%s%s", LINKDEF_PREFIX, id)
+	ldb, err := json.Marshal(ld)
```

Comment:
Super nit: This is more of a personal preference, but we don't need to save bytes. I'd much rather this be something descriptive like `ld_bytes` to make this easier to read on ourselves in the future

Reply:
Sorry, `ldBytes`, because I've been rusting too much

Reply:
you take your 🐍 and leave my 🐫 alone